orderer/common/cluster/clusterservice_test.go

```diff
@@ -0,0 +1,427 @@
+	})
+}
+
+func TestExpirationWarning(t *testing.T) {
+	serverCert, err := ca.NewServerCertKeyPair("127.0.0.1")
+	require.NoError(t, err)
+
+	clientCert, err := ca.NewClientCertKeyPair()
+	require.NoError(t, err)
+
+	dispatcher := &mocks.Dispatcher{}
+	dispatcher.On("DispatchConsensus", mock.Anything, mock.Anything).Return(nil)
+
+	svc := &cluster.ClusterService{
+		CertExpWarningThreshold:          time.Until(clientCert.TLSCert.NotAfter),
+		MinimumExpirationWarningInterval: time.Second * 2,
+		StreamCountReporter: &cluster.StreamCountReporter{
+			Metrics: cluster.NewMetrics(&disabled.Provider{}),
+		},
+		Logger:              flogging.MustGetLogger("test"),
+		StepLogger:          flogging.MustGetLogger("test"),
+		Dispatcher:          dispatcher,
+		MembershipByChannel: make(map[string]map[uint64]*msp.SerializedIdentity),
+	}
+
+	svc.AuthVerifier = &AuthVerifier{}
+
+	alerts := make(chan struct{}, 10)
+	svc.Logger = svc.Logger.WithOptions(zap.Hooks(func(entry zapcore.Entry) error {
+		if strings.Contains(entry.Message, "expires in less than 23h59m") {
+			alerts <- struct{}{}
+		}
+		return nil
+	}))
+
+	srvConf := comm.ServerConfig{
+		SecOpts: comm.SecureOptions{
+			Certificate:       serverCert.Cert,
+			Key:               serverCert.Key,
+			UseTLS:            true,
+			ClientRootCAs:     [][]byte{ca.CertBytes()},
+			RequireClientCert: true,
+		},
+	}
+
+	srv, err := comm.NewGRPCServer("127.0.0.1:0", srvConf)
+	require.NoError(t, err)
+	orderer.RegisterClusterNodeServiceServer(srv.Server(), svc)
+
+	go srv.Start()
+	defer srv.Stop()
+
+	clientConf := comm.ClientConfig{
+		DialTimeout: time.Second * 3,
+		SecOpts: comm.SecureOptions{
+			ServerRootCAs:     [][]byte{ca.CertBytes()},
+			UseTLS:            true,
+			Key:               clientCert.Key,
+			Certificate:       clientCert.Cert,
+			RequireClientCert: true,
+		},
+	}
+
+	conn, err := clientConf.Dial(srv.Address())
+	require.NoError(t, err)
+
+	cl := orderer.NewClusterNodeServiceClient(conn)
+	stream, err := cl.Step(context.Background())
+	require.NoError(t, err)
+
+	err = stream.Send(nodeAuthrequest)
+	require.NoError(t, err)
+	err = stream.Send(nodeConsensusRequest)
+	require.NoError(t, err)
+
+	// An alert is logged at the first time.
+	select {
+	case <-alerts:
+	case <-time.After(time.Second * 5):
+		t.Fatal("Should have received an alert")
+	}
+
+	err = stream.Send(nodeConsensusRequest)
+	require.NoError(t, err)
+
+	// No alerts in a consecutive time.
+	select {
+	case <-alerts:
+		t.Fatal("Should have not received an alert")
+	case <-time.After(time.Millisecond * 500):
+	}
+
+	// Wait for alert expiration interval to expire.
+	time.Sleep(svc.MinimumExpirationWarningInterval + time.Second)
+
+	err = stream.Send(nodeConsensusRequest)
+	require.NoError(t, err)
+
+	// An alert should be logged now after the timeout expired.
+	select {
+	case <-alerts:
+	case <-time.After(time.Second * 5):
+		t.Fatal("Should have received an alert")
+	}
+}
+
```

Comment:
we are still missing tests that spin up a real client and a server in the same unit test, without mocks, just like in `comm_test.go`.

Reply:
Added a new file orderer/common/cluster/commauth_test.go for the real client and server with auth. Please review.